Cura 4.9.1

MathyV opened the pull request:
Version bump (https://bugs.gentoo.org/793803)

This PR also adds python 3.9 to the ebuilds and fixes a potential dependency issue (https://bugs.gentoo.org/731274)

MathyV commented:
Looks like I jumped the gun a bit and missed a new python dependency keyring, checking if I missed any else

MathyV commented:
I added keyring, importlib_metadata and PyQt5-sip since they were added to requirements.txt in https://github.com/Ultimaker/Cura/commit/8bc2c718a2ca0195f20f4f988ae58157c1575c54 I left the other dependencies alone since they got an update.

expeditioneer commented on media-gfx/cura/cura-4.9.1.ebuild:
```diff
@@ -0,0 +1,72 @@
+	~dev-python/uranium-${PV}[${PYTHON_SINGLE_USEDEP}]
+	~media-gfx/curaengine-${PV}
+	~media-gfx/fdm-materials-${PV}
+	dev-qt/qtquickcontrols[widgets]
```
should it not be _dev-qt/qtquickcontrols2_ ? Could you also add the slot to this dependency?

MathyV replied:
tbh I just copied these verbatim from the bug https://bugs.gentoo.org/731274 I will ask the original submitter to comment here

expeditioneer commented on media-gfx/cura/cura-4.9.1.ebuild:
```diff
@@ -0,0 +1,72 @@
+	~media-gfx/curaengine-${PV}
+	~media-gfx/fdm-materials-${PV}
+	dev-qt/qtquickcontrols[widgets]
+	dev-python/PyQt5[-debug]
```
_dev-python/PyQt5_ already listed on Line37

expeditioneer reviewed (changes requested):
Thank you for your contribution. Could you have a look at my review comments?

MathyV commented on dev-libs/libarcus/files/libarcus-4.9.1-deprecated-protobuf-calls.patch:
```diff
@@ -0,0 +1,30 @@
+--- a/src/Socket_p.h
```
My reasoning was that I expect 4.8.0 to disappear soon because of no Python 3.9 in the ebuilds, yet I didn't want to touch the ebuild contents, but okay, I'll adapt it.

MathyV replied:
@expeditioneer is it okay if I rename the file to be versionless and update 4.8.0 also without a revbump?

expeditioneer replied:
@MathyV i would like to have the version number when this patch was first introduced so it is IMHO better to keep track when this patch was first introduced. This is not a strong opinion i could also be happy with a versionless patch.

hrnick commented:
Sorry, that could have been me who did not keep track on the ebuilds in my local repository when patching.

I have not been able to start cura for the past year due to qtquickcontrols not being build with the widgets USE flag (only qtquickcontrols, qtquickcontrols2 does not seem to matter when I tested). While troubleshooting and enabling the debug USE flag for a number of packages I came across the following bug:

https://github.com/Ultimaker/Cura/issues/8875

I have tested that and it's happens for cura-4.9.1 as well.

So sorry for the mistake with having multiple PyQt5s in the ebuild but USE="-debug" has to be used for PyQt5 in order for cura to start.

hrnick commented:
Another comment; I also changed the sip dependency in the libarcus ebuild to use <dev-python/sip-5.5.0 in my local repository since dev-python/sip-5.5.0 seems to break it (ModuleNotFoundError: No module named 'sip' when emerging).

expeditioneer commented:
@hrnick the dependency should be _<dev-python/sip-5[${PYTHON_MULTI_USEDEP}]_ because libarcus is no SIP 5 compatible, see https://github.com/Ultimaker/libArcus/issues/114

MathyV commented:
I updated the ebuilds:
* Removed the duplicate patches
* Merged the two pyqt5 use deps

As I also commented in https://bugs.gentoo.org/795072 there is already a hard dependency on < sip-5 in libarcus so not sure why you are seeing this @hrnick , my best bet is that you are building with ebuild or emerging with --nodeps, on my system portage respects the blocker and downgrades sip.

expeditioneer commented:
@MathyV Thank you for your contribution! I'll merge this in a few seconds.
